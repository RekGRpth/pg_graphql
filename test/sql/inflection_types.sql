begin;
    create view t as
        select
            distinct name
        from
            graphql.type
        where
            entity is not null
        order by
            name;

    create table blog_post(
        id int primary key,
        author_id int
    );

    savepoint a;

    -- Inflection off, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": false})';
    select graphql.rebuild_schema();
    select * from t;

    -- Inflection off, Overrides: on
    comment on table blog_post is e'@graphql({"name": "BlogZZZ"})';
    select graphql.rebuild_schema();
    select * from t;

    rollback to savepoint a;

    -- Inflection on, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": true})';
    select graphql.rebuild_schema();
    select name from graphql.type where entity is not null order by entity, name;

    -- Inflection on, Overrides: on
    comment on table blog_post is e'@graphql({"name": "BlogZZZ"})';
    select graphql.rebuild_schema();
    select name from graphql.type where entity is not null order by entity, name;

rollback;
