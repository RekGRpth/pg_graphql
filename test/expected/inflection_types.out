begin;
    create view t as
        select
            distinct name
        from
            graphql.type
        where
            entity is not null
        order by
            name;
    create table blog_post(
        id int primary key,
        author_id int
    );
    savepoint a;
    -- Inflection off, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": false})';
    select graphql.rebuild_schema();
 rebuild_schema 
----------------
 
(1 row)

    select * from t;
          name           
-------------------------
 blog_post
 blog_postConnection
 blog_postDeleteResponse
 blog_postEdge
 blog_postFilter
 blog_postInsertInput
 blog_postInsertResponse
 blog_postOrderBy
 blog_postUpdateInput
 blog_postUpdateResponse
(10 rows)

    -- Inflection off, Overrides: on
    comment on table blog_post is e'@graphql({"name": "BlogZZZ"})';
    select graphql.rebuild_schema();
 rebuild_schema 
----------------
 
(1 row)

    select * from t;
         name          
-----------------------
 BlogZZZ
 BlogZZZConnection
 BlogZZZDeleteResponse
 BlogZZZEdge
 BlogZZZFilter
 BlogZZZInsertInput
 BlogZZZInsertResponse
 BlogZZZOrderBy
 BlogZZZUpdateInput
 BlogZZZUpdateResponse
(10 rows)

    rollback to savepoint a;
    -- Inflection on, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": true})';
    select graphql.rebuild_schema();
 rebuild_schema 
----------------
 
(1 row)

    select name from graphql.type where entity is not null order by entity, name;
          name          
------------------------
 BlogPost
 BlogPostConnection
 BlogPostDeleteResponse
 BlogPostEdge
 BlogPostFilter
 BlogPostInsertInput
 BlogPostInsertResponse
 BlogPostOrderBy
 BlogPostUpdateInput
 BlogPostUpdateResponse
(10 rows)

    -- Inflection on, Overrides: on
    comment on table blog_post is e'@graphql({"name": "BlogZZZ"})';
    select graphql.rebuild_schema();
 rebuild_schema 
----------------
 
(1 row)

    select name from graphql.type where entity is not null order by entity, name;
         name          
-----------------------
 BlogZZZ
 BlogZZZConnection
 BlogZZZDeleteResponse
 BlogZZZEdge
 BlogZZZFilter
 BlogZZZInsertInput
 BlogZZZInsertResponse
 BlogZZZOrderBy
 BlogZZZUpdateInput
 BlogZZZUpdateResponse
(10 rows)

rollback;
